dbms/src/Flash/ResourceControl/MockLocalAdmissionController.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/ResourceControl/LocalAdmissionController.h>
+
+namespace DB
+{
+void MockLocalAdmissionController::refillTokenBucket()
+{
+    while (true)
+    {
+        {
+            std::unique_lock lock(mu);
+            if (cv.wait_for(lock, std::chrono::seconds(1), [&]() { return stopped; }))
+                return;
+
+            for (auto & ele : resource_groups)
+            {
+                auto & rg = ele.second;
+                if (rg->bucket->isStatic())
+                {
+                    std::cout << "refill token bucket of rg " << rg->name << std::endl;
+                    const auto bucket_config = rg->bucket->getCurrentConfig();
+                    const double new_fill_rate = std::get<1>(bucket_config);
+                    const double new_capacity = std::get<2>(bucket_config);
+                    // Use capacity as new token, because for static token bucket, its fill rate is zero.
+                    const double new_token = rg->bucket->peek() + std::get<2>(bucket_config);
```

Comment:
```suggestion
                    std::tie(new_fill_rate, new_capacity, new_token) = rg->bucket->getCurrentConfig();
                    // Use capacity as new token, because for static token bucket, its fill rate is zero.
                    new_token += rg->bucket->peek()
```

Reply:
changed to structured binding